### PR TITLE
20200207 update webauthn api

### PR DESCRIPTION
**Pull request by Firstyear**
Implements #19 improve api ergonomics .

- [ x ] cargo fmt has been run
- [ x ] cargo test has been run and passes
- [ x ] documentation has been updated with relevant examples (if relevant)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -101,7 +113,7 @@ impl<T> Webauthn<T> {
         Webauthn {
             // We use stdrng because unlike thread_rng, it's a csprng, which given
```
Comment (agrinman) [inline src/lib.rs]: Outdated comment?
`thread_rng` is cryptographically secure and it is what's being used.

File: src/lib.rs [hunk 2]
```diff
@@ -509,27 +536,90 @@ impl<T> Webauthn<T> {
...
+    ///
+    /// You must persist the AuthenticationState that is returned. You should associate this by
+    /// UserId. The AuthenticationState is required for the authenticate_credential function to
+    /// operate correctly.
+    ///
+    /// At this time we deviate from the standard and base64 some fields, but we are
+    /// investigating how to avoid this (https://github.com/Firstyear/webauthn-rs/issues/5)
+    pub fn generate_challenge_authenticate(
+        &mut self,
+        _username: &UserId,
```
Comment (agrinman) [inline src/lib.rs]: It looks like this field is unnecessary?

File: src/lib.rs (hunk 1 above)
Comment (Firstyear) [reply src/lib.rs]: Fixed

File: src/lib.rs (hunk 2 above)
Comment (Firstyear) [reply src/lib.rs]: And removed :)

**Pull request merged**